tools/mac-distribution/notarize/notarize.go

```diff
@@ -0,0 +1,221 @@
+package notarize
+
+import (
+	"log/slog"
+	"os"
+	"path/filepath"
+
+	"github.com/gravitational/shared-workflows/tools/mac-distribution/internal/exec"
+	"github.com/gravitational/shared-workflows/tools/mac-distribution/internal/fileutil"
+	"github.com/gravitational/shared-workflows/tools/mac-distribution/internal/zipper"
+
+	"github.com/gravitational/trace"
+)
+
+// Tool is a wrapper around the MacOS codesigning/notarizing utilities.
+type Tool struct {
+	Creds Creds
+	// Retry sets the max number of attempts for a succesful notarization
+	retry int
```

Comment:
```suggestion
	maxRetries int
```

A well-named variable often eliminates the need for extra comments.